Gems/Atom/RPI/Code/Source/RPI.Reflect/Image/StreamingImageAssetHandler.cpp

```diff
@@ -36,8 +136,32 @@ namespace AZ
                 if (loadResult == Data::AssetHandler::LoadResult::LoadComplete)
                 {
                     // ImageMipChainAsset has some internal variables need to initialized after it was loaded.
-                    StreamingImageAsset* assetData2 = asset.GetAs<StreamingImageAsset>();
-                    assetData2->m_tailMipChain.Init();
+                    assetData->m_tailMipChain.Init();
+
+                    // Find assets for all the mipchains
+                    // This is to resolve reloading StreamingImageAsset won't load the asset reference even the StreamingImageAsset was loaded with LoadAll option
+                    // If the asset was loaded. Check if it has full mipchain loaded.
+                    auto& assetManager = AZ::Data::AssetManager::Instance();
+                    Data::Asset<StreamingImageAsset> foundImageAsset = assetManager.FindAsset(asset.GetId(), AZ::Data::AssetLoadBehavior::PreLoad);
+                    if (foundImageAsset && foundImageAsset->HasFullMipChainAssets())
+                    {
+                        for (auto& mipChainAsset : assetData->m_mipChains)
+                        {
+                            if (mipChainAsset.m_asset.GetId().IsValid() && !mipChainAsset.m_asset.Get())
+                            {
+                                if (auto foundAsset = assetManager.FindAsset(mipChainAsset.m_asset.GetId(), AZ::Data::AssetLoadBehavior::PreLoad))
+                                {
+                                    mipChainAsset.m_asset = foundAsset;
+                                }
+                                else
+                                {
+                                    // This may happen if the image's resolution increased which generate new mipchain assets 
+                                    mipChainAsset.m_asset.QueueLoad();
+                                    mipChainAsset.m_asset.BlockUntilLoadComplete();
```

Comment:
Ideally this would queue all the loads in the for loop, then have a separate for loop with the BlockUntilLoadComplete, so that way the loads can happen in parallel.